### PR TITLE
Add mffdiff

### PR DESCRIPTION
**Pull request by jusjusjus**
I've added a small tool here to compare the content of two MFF files.  It's not quite complete yet, but will be useful enough to test the output of MFF post-processing tools that write back to MFF.

### CODE REVIEW

File: bin/mffdiff.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+compare('startdatetime')
+compare('units')
+compare('num_channels')
+compare('categories.categories')
+epoch_count = compare('epochs.epochs', lambda e: len(e), 'len(..)')
+if epoch_count is not None:
+    for i in range(epoch_count):
+        compare('epochs.epochs', lambda e: e[i].content, f"[{i}].content")
+
+out = compare_raw('signal1.bin')
```
Comment (damian5710) [inline bin/mffdiff.py:83]: Should we move these lines to a function that returns the result of the comparison (i.e. whether the files match or not)?

File: bin/mffdiff.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+""")
+parser.add_argument('leftpath', type=str, help="MFF file to compare")
+parser.add_argument('rightpath', type=str, help="second MFF file to compare")
+args = parser.parse_args()
+
+left_mff = Reader(args.leftpath)
+right_mff = Reader(args.rightpath)
+
+
+def getnested(cls, props, callback=None):
```
Comment (damian5710) [inline bin/mffdiff.py:23]: Docstrings are missing in these functions.

File: bin/mffdiff.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    print(f">>> {status} @ file '{filename}'")
+
+
+def compare(prop, callback=None, info=''):
+    props = prop.split('.')
+    try:
+        left_value = getnested(left_mff, props, callback)
+        right_value = getnested(right_mff, props, callback)
+        left_msg = str(left_value)[:50]
+        right_msg = str(right_value)[:50]
```
Comment (damian5710) [inline bin/mffdiff.py:49]: Where does the `50` come from?

File: bin/mffdiff.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+
+print(f"Comparing {args.leftpath} with {args.rightpath}")
+compare('flavor')
+compare('sampling_rates')
+compare('durations')
+compare('startdatetime')
+compare('units')
+compare('num_channels')
+compare('categories.categories')
```
Comment (damian5710) [inline bin/mffdiff.py:77]: We could create a list for these props and iterate through it using a for loop.

**Review by damian5710 — CHANGES_REQUESTED** [review]
This is a great start. I tested this new script with some mff files and I think it is working as expected.

File: bin/mffdiff.py (hunk 1 above)
Comment (jusjusjus) [reply bin/mffdiff.py]: Probably not needed.

**Review by damian5710 — APPROVED** [review]

**Pull request merged**